rpcserverhandlers_test.go

```diff
@@ -6,28 +6,854 @@ package main
+			PkScript: []byte{
+				0xBD, 0x76, 0xA9, 0x14, 0x00, 0x00, 0x00, 0x00,
+				0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+				0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+				0x88, 0xAC,
+			},
+			Value:   0,
+			Version: 0,
+		}}
+	default:
+		return []*stake.MinimalOutput{{
+			PkScript: []byte{
+				0xBA, 0xA9, 0x14, 0x78, 0x02, 0x39, 0xEA, 0x12,
+				0x31, 0xBA, 0x67, 0xB0, 0xC5, 0xB8, 0x2E, 0x78,
+				0x6B, 0x51, 0xE2, 0x10, 0x72, 0x52, 0x21, 0x87,
+			},
+			Value:   100000000,
+			Version: 0,
+		}, {
+			PkScript: []byte{
+				0x6A, 0x1E, 0x35, 0x5C, 0x96, 0xF4, 0x86, 0x12,
+				0xD5, 0x75, 0x09, 0x14, 0x0E, 0x9A, 0x04, 0x99,
+				0x81, 0xD5, 0xF9, 0x97, 0x0F, 0x94,
+				0x5C, 0x77, 0x0D, 0x00, 0x00, 0x00, 0x00, 0x00, // commitamt
+				0x00, 0x58,
+			},
+			Value:   0,
+			Version: 0,
+		}, {
+			PkScript: []byte{
+				0xBD, 0x76, 0xA9, 0x14, 0x00, 0x00, 0x00, 0x00,
+				0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+				0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+				0x88, 0xAC,
+			},
+			Value:   0,
+			Version: 0,
+		}}
+	}
+}
+
+// FetchUtxoEntry returns a mocked UtxoEntry based on the provided txHash
+func (c *testRPCChain) FetchUtxoEntry(txHash *chainhash.Hash) (rpcserver.UtxoEntry, error) {
+	c.testTxHash = txHash
+	var utxoEntry rpcserver.UtxoEntry
+	switch c.testTxHash.String() {
+	case "1189cbe656c2ef1e0fcb91f107624d9aa8f0db7b28e6a86f694a4cf49abc5e39",
+		"3189cbe656c2ef1e0fcb91f107624d9aa8f0db7b28e6a86f694a4cf49abc5e39",
+		"4189cbe656c2ef1e0fcb91f107624d9aa8f0db7b28e6a86f694a4cf49abc5e39":
+		utxoEntry = &testRPCUtxoEntry{
+			txType:     stake.TxTypeSStx,
+			height:     100000,
+			index:      0,
+			txVersion:  1,
+			isCoinBase: false,
+			hasExpiry:  true,
+			modified:   false,
+		}
+	case "2189cbe656c2ef1e0fcb91f107624d9aa8f0db7b28e6a86f694a4cf49abc5e39":
+		utxoEntry = &testRPCUtxoEntry{
+			txType:     stake.TxTypeRegular,
+			height:     100000,
+			index:      0,
+			txVersion:  1,
+			isCoinBase: false,
+			hasExpiry:  true,
+			modified:   false,
+		}
+	}
+	return utxoEntry, nil
+}
+
+// FetchUtxoStats returns a mocked blockchain.UtxoStats
+func (c *testRPCChain) FetchUtxoStats() (*blockchain.UtxoStats, error) {
+	hash, _ := chainhash.NewHashFromStr("fe7b32aa188800f07268b17f3bead5f3d8a1b6d18654182066436efce6effa86")
+	return &blockchain.UtxoStats{
+		Utxos:          1593879,
+		Transactions:   689819,
+		Size:           36441617,
+		Total:          1154067750680149,
+		SerializedHash: *hash,
+	}, nil
+}
+
 // testCfg holds the cfg for the test server.
-var testCfg = rpcserverConfig{ChainParams: chaincfg.MainNetParams()}
+var testCfg = rpcserverConfig{
+	ChainParams: chaincfg.MainNetParams(),
+	Chain:       &testRPCChain{},
+}
 
 // testServer is a rpcServer stub used for testing handlers.
 var testServer = &rpcServer{cfg: testCfg}
 
+type rpcTest struct {
+	name    string
+	handler commandHandler
+	cmd     interface{}
+	result  interface{}
+	wantErr bool
+	errCode dcrjson.RPCErrorCode
+}
```

Comment:
Yep, I noticed that as well - but in this case, the struct type is used across all tests, so I thought it made more sense to define the type once rather than defining the type inline in every test.

Reply:
I do very much prefer inline test structs for consistency, but there are a couple of other instances where it isn't true due to the same reason as here.  Given every one of these tests are identical, it isn't unreasonable to have the struct defined separately in this case.